src/process_utility.c

```diff
@@ -4482,6 +4482,21 @@ process_create_trigger_start(ProcessUtilityArgs *args)
 				 errmsg("ROW triggers with transition tables are not supported on hypertables")));
 	}
 
+	/*
+	 * We currently cannot support delete triggers with transition tables on
+	 * compressed tables that are not using hypercore table access method
+	 * since deleting a complete segment will not build a transition table for
+	 * the delete.
+	 */
+	if (stmt->transitionRels && TRIGGER_FOR_DELETE(tgtype) &&
+		TS_HYPERTABLE_HAS_COMPRESSION_ENABLED(ht) && !ts_is_hypercore_am(ht->amoid))
+	{
+		ts_cache_release(hcache);
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("DELETE triggers with transition tables not supported for this case")));
```

Comment:
I think we need to rephrase a bit the error message to make clear that we don't support DELETE triggers with transition tables on compressed hypertables...  the wordking `... not supported for this case` is a bit vague to me.

Reply:
I agree it is vague. If you have a suggestion, I'm all game.

Reply:
I changed it according to our discussions.